plugins/cassandra/cassandraembed.cpp

```diff
@@ -4225,14 +4734,218 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
     }
     virtual IConstWorkUnitIterator * getScheduledWorkUnits(ISecManager *secmgr, ISecUser *secuser)
     {
-        return getWorkUnitsByXXX("@state", getWorkunitStateStr(WUStateScheduled), secmgr, secuser);
+        return getWorkUnitsByXXX("@state", getWorkunitStateStr(WUStateScheduled), secmgr, secuser); // MORE - there may be more efficient ways to do this?
     }
     virtual IConstWorkUnitIterator * getWorkUnitsSorted(WUSortField sortorder, WUSortField * filters, const void * filterbuf,
-                                                        unsigned startoffset, unsigned maxnum, const char * queryowner, __int64 * cachehint, unsigned *total,
+                                                        unsigned startOffset, unsigned pageSize, __int64 * cachehint, unsigned *total,
                                                         ISecManager *secmgr, ISecUser *secuser)
     {
-        // Note that we only support a single sort order - which makes life easier!
-        UNIMPLEMENTED;
+        // Note that we only support a single sort order.
+        // Any sort order other than WUID ASC or WUID DESC will require post-sorting - if there are more than WUID_LOCALSORT_LIMIT we will refuse
```

Comment:
Should get a bigger screen!

I'm slightly torn about the long lines issue - it does make reviewing harder, but breaking lines up can make reading the code later harder (assuming you are doing so on a sensible screen. It's a shame GitHub does not show a little wider.